test/test_resume.cpp

```diff
@@ -192,6 +192,58 @@ void default_tests(torrent_status const& s)
 	TEST_EQUAL(s.completed_time, 1348);
 }
 
+void test_file_sizes(bool allocate)
+{
+	error_code ec;
+	remove_all("test_resume", ec);
+
+	lt::settings_pack pack = settings();
+	// we're not testing the hash check, just accept the data we write
+	pack.set_bool(settings_pack::disable_hash_checks, true);
+	lt::session ses(pack);
+	boost::shared_ptr<torrent_info> ti = generate_torrent();
+	add_torrent_params p;
+	p.ti = ti;
+	p.save_path = ".";
+	if (allocate) p.storage_mode = storage_mode_allocate;
+	torrent_handle h = ses.add_torrent(p);
+
+	std::vector<char> piece(ti->piece_length(), 0);
+	h.add_piece(0, piece.data());
```

Comment:
before saving resume data, we need to synchronize with the piece actually being written here. by waiting for the ``piece_finished_alert``